crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -227,6 +226,21 @@ async fn validate_op_inner(
         DhtOp::StoreElement(_, header, entry) => {
             store_element(header, workspace, network.clone()).await?;
             if let Some(entry) = entry {
+                // Retrieve for all other headers on countersigned entry.
+                if let Entry::CounterSign(session_data, _) = &**entry {
+                    for header in session_data.build_header_set()? {
+                        workspace
+                            .local_cascade()
+                            .retrieve(
+                                HeaderHashed::from_content_sync(header.clone())
```

Comment:
Just do `HeaderHash::with_data_sync(&header)` then you don't need to clone the header to get the hash